### PR TITLE
Implement broadcast_to

### LINKED ISSUE #642: Support `broadcast_to`
http://docs.scipy.org/doc/numpy/reference/generated/numpy.broadcast_to.html
I now make a dummy variable...

### PR DESCRIPTION
**Pull request by unnonouno**
fix #642

### CODE REVIEW

**Comment by okuta** [comment]
+1

File: cupy/manipulation/dims.py [hunk 1]
```diff
@@ -122,6 +122,41 @@ def broadcast_arrays(*args):
     return broadcast(*args).values
 
 
+def broadcast_to(array, shape):
+    """Broadcast an array ot a given shape.
```
Comment (beam2d) [inline cupy/manipulation/dims.py]: typo (ot -> to)

File: cupy/manipulation/dims.py [hunk 2]
```diff
@@ -122,6 +122,41 @@ def broadcast_arrays(*args):
...
+
+    Args:
+        array (cupy.ndarray): Array to broadcast.
+        shape (tuple of int): The shape to desired array.
```
Comment (beam2d) [inline cupy/manipulation/dims.py]: doc -> The shape of the desired array

File: cupy/manipulation/dims.py [hunk 3]
```diff
@@ -122,6 +122,41 @@ def broadcast_arrays(*args):
...
+            'remapping')
+
+    strides = [0] * len(shape)
+    for i in range(array.ndim):
+        j = -i - 1
+        sh = shape[j]
+        a_sh = array.shape[j]
+        if sh == a_sh:
+            strides[j] = array._strides[j]
+        elif a_sh == 1:
```
Comment (beam2d) [inline cupy/manipulation/dims.py]: How about reducing these elif/else into `elif a_sh != 1: raise ...`?

File: tests/cupy_tests/manipulation_tests/test_dims.py [hunk 4]
```diff
@@ -65,6 +65,54 @@ def test_broadcast_fail(self, xp):
...
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_raises()
+    def test_broadcast_to_short_shape(self, xp, dtype):
+        # Note that broadcast_to is only supported on numpy>=1.10
+        a = testing.shaped_arange((1, 3, 4), xp, dtype)
+        xp.broadcast_to(a, (3, 4))
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal(accept_error=False)
+    def test_broadcast_to_19(self, xp, dtype):
```
Comment (beam2d) [inline tests/cupy_tests/manipulation_tests/test_dims.py]: What is 19?

**Comment by beam2d** [comment]
I added some comments.

File: tests/cupy_tests/manipulation_tests/test_dims.py (hunk 4 above)
Comment (unnonouno) [inline tests/cupy_tests/manipulation_tests/test_dims.py]: Test for numpy 1.9

**Comment by unnonouno** [comment]
I fixed it.

**Comment by beam2d** [comment]
LGTM! Can you add it to the reference manual?

**Comment by beam2d** [comment]
[LGTM](http://lgtm.in/i/O1Cmdslqq)

**Pull request merged**